libraries/eosiolib/core/eosio/crypto_bls_ext.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include "fixed_bytes.hpp"
+#include "varint.hpp"
+#include "serialize.hpp"
+
+#include <array>
+#include <vector>
+
+namespace eosio {
+
+    namespace internal_use_do_not_use {
+        extern "C" {
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g1_add(const char* op1, uint32_t op1_len, const char* op2, uint32_t op2_len, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g2_add(const char* op1, uint32_t op1_len, const char* op2, uint32_t op2_len, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g1_mul(const char* point, uint32_t point_len, const char* scalar, uint32_t scalar_len, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g2_mul(const char* point, uint32_t point_len, const char* scalar, uint32_t scalar_len, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g1_exp(const char* points, uint32_t points_len, const char* scalars, uint32_t scalars_len, uint32_t n, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g2_exp(const char* points, uint32_t points_len, const char* scalars, uint32_t scalars_len, uint32_t n, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_pairing(const char* g1_points, uint32_t g1_points_len, const char* g2_points, uint32_t g2_points_len, uint32_t n, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g1_map(const char* e, uint32_t e_len, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_g2_map(const char* e, uint32_t e_len, char* res, uint32_t res_len);
+
+            __attribute__((eosio_wasm_import))
+            int32_t bls_fp_mod(const char* s, uint32_t s_len, char* res, uint32_t res_len);
+        }
+    }
+
+    int32_t bls_g1_add(const uint8_t* op1, const uint8_t* op2, uint8_t* res)
```

Comment:
I think it is better to have something like this:
```c++
int32_t bls_g1_add(const uint8_t (&op1)[144], const (&op2)[144], uint8_t (&res)[144])
```
also, you could typedef those and it will be even nicer:
```c++
using some_type = uint8_t[144];
    int32_t bls_g1_add(const some_type& op1, const some_type& op2, some_type& res)
```

Reply:
Added bls types to enforce correct operand sizes in: https://github.com/mschoenebeck/cdt/commit/9d1424dfcf038cbaa096dda1976fcc44c0443adb